contracts/ethereum_contracts.go

```diff
@@ -23,11 +25,152 @@ type EthereumFluxAggregator struct {
 func (f *EthereumFluxAggregator) Fund(fromWallet client.BlockchainWallet, ethAmount, linkAmount *big.Int) error {
 	return f.client.Fund(fromWallet, f.address.Hex(), ethAmount, linkAmount)
 }
 
+func (f *EthereumFluxAggregator) UpdateAvailableFunds(ctx context.Context, fromWallet client.BlockchainWallet) error {
+	opts, err := f.client.TransactionOpts(fromWallet, *f.address, big.NewInt(0), nil)
+	if err != nil {
+		return err
+	}
+	tx, err := f.fluxAggregator.UpdateAvailableFunds(opts)
+	if err != nil {
+		return err
+	}
+	if err := f.client.WaitForTransaction(tx.Hash()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (f *EthereumFluxAggregator) PaymentAmount(ctx context.Context) (*big.Int, error) {
+	opts := &bind.CallOpts{
+		From:    common.HexToAddress(f.callerWallet.Address()),
+		Pending: true,
+		Context: ctx,
+	}
+	payment, err := f.fluxAggregator.PaymentAmount(opts)
+	if err != nil {
+		return nil, err
+	}
+	return payment, nil
+}
+
+func (f *EthereumFluxAggregator) RequestNewRound(ctx context.Context, fromWallet client.BlockchainWallet) error {
+	opts, err := f.client.TransactionOpts(fromWallet, *f.address, big.NewInt(0), nil)
+	if err != nil {
+		return err
+	}
+	tx, err := f.fluxAggregator.RequestNewRound(opts)
+	if err != nil {
+		return err
+	}
+	if err := f.client.WaitForTransaction(tx.Hash()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (f *EthereumFluxAggregator) SetRequesterPermissions(ctx context.Context, fromWallet client.BlockchainWallet, addr common.Address, authorized bool, roundsDelay uint32) error {
+	opts, err := f.client.TransactionOpts(fromWallet, *f.address, big.NewInt(0), nil)
+	if err != nil {
+		return err
+	}
+	tx, err := f.fluxAggregator.SetRequesterPermissions(opts, addr, authorized, roundsDelay)
+	if err != nil {
+		return err
+	}
+	return f.client.WaitForTransaction(tx.Hash())
+}
+
+func (f *EthereumFluxAggregator) GetOracles(ctx context.Context) ([]string, error) {
+	opts := &bind.CallOpts{
+		From:    common.HexToAddress(f.callerWallet.Address()),
+		Pending: true,
+		Context: ctx,
+	}
+	addresses, err := f.fluxAggregator.GetOracles(opts)
+	if err != nil {
+		return nil, err
+	}
+	var oracleAddrs []string
+	for _, o := range addresses {
+		oracleAddrs = append(oracleAddrs, o.Hex())
+	}
+	return oracleAddrs, nil
+}
+
+func (f *EthereumFluxAggregator) LatestRound(ctx context.Context) (*big.Int, error) {
+	opts := &bind.CallOpts{
+		From:    common.HexToAddress(f.callerWallet.Address()),
+		Pending: true,
+		Context: ctx,
+	}
+	rID, err := f.fluxAggregator.LatestRound(opts)
+	if err != nil {
+		return nil, err
+	}
+	return rID, nil
+}
+
+// AwaitNextRound awaits for the next round to happen, see config.yaml for default values
+func (f *EthereumFluxAggregator) AwaitNextRound(ctx context.Context) error {
+	lr, err := f.LatestRound(ctx)
+	if err != nil {
+		return err
+	}
+	log.Info().Int64("round", lr.Int64()).Msg("awaiting next round after")
+	if err := retry.Do(func() error {
```

Comment:
Love the retry mechanism, I'm just wondering if this logic would be better to be converted to a log subscription and timeout so it doesn't have to poll?

Reply:
Tried log subscription, I think polling is ok in that case because we need to know when the round is finalized, not started.
Will do event awaiting and logging in a separate PR.